index.tsx

```diff
@@ -112,12 +112,12 @@ export const html = (text: string, opts = {}) => {
   unimplemented('html export');
 };
 
-const astProcessor = (opts = {}) =>
+const astProcessor = (opts: MdastOpts = { components: {} }) =>
```

Comment:
Can `components` get more specific or is it just `Record<string, string>`?

Reply:
I don't think so? It's a mapping of tag name to component body?

Reply:
might want to narrow it to `Record<string, string>` because right now it'll allow `Record<string, unknown>`

Reply:
How?

![image](https://github.com/readmeio/markdown/assets/451488/d8b06dd9-c3e3-49ad-8241-07338580820d)

Reply:
oh my bad. `MdastOpts` is where `components` is defined, I misread the `=` there as `:` and thought you were typing `opts: { components: {} }`

Reply:
Thought I was about to have my type system shattered.

Reply:
nope im just a dumbass. carry on